### ISSUE TITLE
Selectonemenu: Limited accessibility of ui-selectonemenu-list with Screenreader Jaws 17/ Internet Explorer 11

### ISSUE THREAD

**Issue by HISeG-ME** [primefaces/primefaces#4660]
## 1) Environment
- Primefaces Version: 6.2
- Affected Browsers: Internet Explorer 11/ Jaws 17

## 2) Expected behavior
When using the Screenreader Jaws, the user will tab into the Selectonemenu (advanced). If it gets the focus, the List can be opened with the Spacer bar. Thereafter, the Inputfield will get the focus and its possible to navigate through the available Listitems using the arrow up/ down keys.
...

## 3) Actual behavior
When the Inputfield gets the focus, its possible to navigate through the Listitems using the arrow down key. When using the arrow up key, the focus will leave the List. Due to the Selectonemenurenderer, the List is rendered as <ul> with the Listitems as <li> Elements, since we don´t use customcontent (implemented in the Selectonemenurenderer.java).
The Primefaces Showcase page renders the Listitems as <tr> and <td> Elements.
..

## 4) Steps to reproduce
 Visit the page https://www.primefaces.org/showcase/ui/input/oneMenu.xhtml 
using the Screenreader Jaws 17 with the Browser Internet Explorer 11. When tabbing in the advanced Selectonemenu and open it with the Space bar, the arrow up/ down keys can be used to navigate the List. The Screenreader will read the Images and values of the Tablecells and won´t leave the List.
.. 

## 5) Sample XHTML
Actual rendered List in our environment looks like this:

<div class="ui-selectonemenu-items-wrapper" style="max-height:200px">
<ul id="selbstregistrierung:persondata:gender_items" class="ui-selectonemenu-items ui-selectonemenu-list ui-widget-content ui-widget ui-corner-all ui-helper-reset" role="listbox" aria-activedescendant="selbstregistrierung:persondata:gender_0">
<li class="ui-selectonemenu-item ui-selectonemenu-list-item ui-corner-all ui-state-highlight" data-label="&nbsp;" tabindex="-1" role="option" id="selbstregistrierung:persondata:gender_0">&nbsp;</li>
<li class="ui-selectonemenu-item ui-selectonemenu-list-item ui-corner-all" data-label="männlich" tabindex="-1" role="option" id="selbstregistrierung:persondata:gender_1">männlich</li>
<li class="ui-selectonemenu-item ui-selectonemenu-list-item ui-corner-all" data-label="undefiniert / k.A." tabindex="-1" role="option" id="selbstregistrierung:persondata:gender_2">undefiniert / k.A.</li>
<li class="ui-selectonemenu-item ui-selectonemenu-list-item ui-corner-all" data-label="weiblich" tabindex="-1" role="option" id="selbstregistrierung:persondata:gender_3">weiblich</li>
</ul>
</div>

..

## 6) Sample bean

..

**Comment by melloware** [comment]
Is this specific to IE and JAWS17 or does it work ok with other browsers?

**Comment by HISeG-ME** [comment]
In Firefox and Chrome, Jaws allows the arrow navigation but won´t read the Listitems unless they´re selected and appear in the Inputfield (Chrome) and won´t be read at all (Firefox).

**Comment by melloware** [comment]
@HISeG-ME If you can look at the HTML source  in the browser and determine what we are missing or what ARIA tags may be in the wrong places that is causing this issue I definitely don't mind fixing it.  But typically I count on people who actually use these screenreaders to help us figure out what we need to change.  Just let me know.

**Comment by melloware** [comment]
I dug more into this.  Every time you arrow up or arrow down it calls highlightNext or HighlightPrevious. In that code it calls this ine.

https://github.com/primefaces/primefaces/blob/master/src/main/resources/META-INF/resources/primefaces/forms/forms.selectonemenu.js#L594

and that changeAriaValue method looks like its doing the right thing when I watch the DOM.

```javascript
changeAriaValue: function (item) {
        var itemId = item.attr('id');

        this.focusInput.attr('aria-activedescendant', itemId) .attr('aria-describedby', itemId);
        this.itemsContainer.attr('aria-activedescendant', itemId);
    },
```

**Comment by melloware** [comment]
@MarkusChur can you look at this one and explain it to me?

**Comment by MarkusChur** [comment]
yes, SelectOneMenu has a few issue though. I'll take a look and provide you a list with suggestions later this week.

**Comment by MarkusChur** [comment]
I have used the recent show case for my tests.

Core problem is that the combobox and the list lack of attributes so that these can be interpreted correctly by assistive technology. Not only the advanced combox is affected.
* Unable to use the list with voice commands
* Unable to know possible options as a screen reader only user
* Text Input is always dominant and read out even if keyboard focus is on list items. For example if input value is empty and you traverse the list with the keyboard screenreaders read out "blank" in any case.
* No way to know the amount of list items as a screen reader only user

Consider https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html and Example 3.
There is an explanation of roles and on which components the attributes have to be applied.

As far as I can see:
* `aria-haspopup="true"` is used on div combobox container, I think it should it be `aria-haspopup="listbox"` 
* `aria-owns=""` is missing on div combobox container. It should own the advanced panel id I think.
* text input is missing `aria-activedescendant` it is currently applied to the table of items. The input is focussed so I think when it's not applied here active selection is not recognized.
* text input `aria-autocomplete="list"` is missing
* text input is missing `aria-controls="table_list_id"`
* Table of list items is missing `id`
* Table of list item is missing `aria-labelledby`
* Table of list items is missing `role="listbox"`
* Table rows are missing `role="option"`
* Table rows are missing `aria-selected="true"`

**Comment by melloware** [comment]
@MarkusChur I checked in both Splitbutton and SelectOne ARIA updates.  If you would like to test them you can use the JitPack build locally.

```xml
<dependency>
     <groupId>com.github.primefaces</groupId>
     <artifactId>primefaces</artifactId>
     <version>master-SNAPSHOT</version>
 </dependency>

 <repositories>
 	<repository>
 	    <id>jitpack.io</id>
 	    <url>https://jitpack.io</url>
 	</repository>
 </repositories>
```
